File: lv-vhdl-frontend/src/main/java/stupaq/translation/ExceptionPrinter.java
package stupaq.translation;

import java.io.PrintStream;

import stupaq.vhdl93.ParseException;

public final class ExceptionPrinter {
  private ExceptionPrinter() {
  }

  public static void print(Exception exception, PrintStream stream) {
    exception.printStackTrace();
    if (exception instanceof SemanticException) {
      stream.println("Translation error encountered:");
      stream.println(exception.getMessage());
    } else if (exception instanceof MissingFeatureException) {
      stream.println("Translation error encountered:");
      stream.println(exception.getMessage());
    } else if (exception instanceof ParseException) {
      stream.println("Translation error encountered:");
      stream.println(exception.getMessage());
    } else {
      exception.printStackTrace(stream);
    }
  }
}

File: lv-vhdl-frontend/src/main/scala/stupaq/translation/ExceptionPrinter.scala
package stupaq.translation

import java.io.PrintStream

import stupaq.translation.errors.TranslationException
import stupaq.translation.lv2vhdl.LocalisedException

object ExceptionPrinter {

  def print(ex: Exception, stream: PrintStream) {
    ex match {
      case _: TranslationException =>
        stream println "Translation error encountered:"
        stream println ex.getMessage
      case _ =>
        ex printStackTrace stream
    }
  }
}

